app/scripts/controllers/reviewRegistration.js

```diff
@@ -68,6 +68,105 @@ angular.module('confRegistrationWebApp')
+    }
+
+    //Click event for 'Include Spouse' button
+    $scope.includeSpouseInRegistration = function () {
+
+      //Parameters needed to retrieve all current registrants for conference.id
+      var queryParameters = {
+        block: [],
+        //page: 1,
+        //limit: 20,
+        orderBy: 'last_name',
+        order: 'ASC',
+        filter: '',
+        filterPayment: '',
+        filterRegType: '',
+        includeCheckedin: 'yes',
+        includeWithdrawn: 'yes',
+        includeIncomplete: 'yes'
+      };
+
+      //If false then spouse found has not been regitered yet
+      var spouseAlreadyRegistered = false;
+
+      //If spouse does not exists in registry then enter a new registrant based on return from endPointSpouseInfo
+      //else no need to register because already registered
+      var updateRegistrants = function() {
+        if (!spouseAlreadyRegistered) {
+          var registrationId = uuid();
+          $scope.currentRegistration.registrants.push({
+            id: uuid(),
+            registrationId: registrationId,
+            registrantTypeId: $scope.currentRegistration.registrants[0].registrantTypeId,
+            firstName: endPointSpouseInfo.firstName,
+            lastName: endPointSpouseInfo.lastName,
+            email: endPointSpouseInfo.email,
+            answers: [{id: uuid(), registrantId: registrationId, blockId: $scope.currentRegistration.registrants[0].answers[0].blockId, value: {firstName: endPointSpouseInfo.firstName, lastName: endPointSpouseInfo.lastName}},
+              {id: uuid(), registrantId: registrationId, blockId: $scope.currentRegistration.registrants[0].answers[1].blockId, value: endPointSpouseInfo.email}]
+          });
+          RegistrationCache.update('registrations/' + $scope.currentRegistration.id, $scope.currentRegistration, function () {
+            $route.reload();
+          }, function(){
+            alert('An error occurred while updating your registration (2).');
+          });
+        }
+      };
+
+      //Get all current registrants based on conference.id and queryParameters
+      RegistrationCache.getAllForConference(conference.id, queryParameters).then(function (data) {
+        for (var i = 0; i < data.registrations.length; i++) {
+          if (endPointSpouseInfo.email === data.registrations[i].registrants[0].email) {
+            spouseAlreadyRegistered = true;
+            //$scope.currentRegistration.registrants.push(data.registrations[i].registrants[0]);
+            //RegistrationCache.updateCurrent(conference.id, $scope.currentRegistration);
+            alert('Registration will be included with spouse.');
+          }
+        }
+        updateRegistrants();
+      }, function () {
+        alert('An error occurred while updating your registration (1).');
+      });
+
+      //Control whether or not 'Include Spouse' button is hidden or not
+      localStorage.setItem('includeSpouseButtonVisible', 'false');
```

Comment:
I agree. I was bit hesitant to use it initially for the same reason.  I'll see if there is a better solution.